share/wake/lib/system/io.wake

```diff
@@ -187,6 +193,16 @@ export def write (path: String) (content: String): Result Path Error =
 
     writeImp (parent, Nil) 0644 spath content
 
+# Writes a string to an intermediate file that will not conflict with other files.
+#
+# Idential prefix+files will be automatically deduplicated
+export def writeTempFile (prefix: String) (content: String): Result Path Error =
```

Comment:
For my understanding: If this were a `target` would the deduplication happen faster? Would it be a stronger deduplication guarantee (exact input equality vs hash(input) equality)?

Reply:
`writeIn` bottoms out in a `target` so there is a couple tradeoffs being made here.

making `writeTempFile` a target will 
 - skip the `hashStr` hash
 - hash prefix & content (the runtime needs to hash them to look it up in the memo table)
 - have two hash tables with nearly duplicate data (`writeIn` would have an entry and `writeTempFile` would have an entry)
making `writeTempFile` a def will
  - calculate the `hashStr` hash every time 
  - not hash prefix & content params
  - only have one entry in the `writeIn` memo table
  
  I'm happy either way but to me a `def` seems like slightly less overhead

Reply:
yeah the current implementation sounds like the right tradeoff to me. Thanks for the education on how this bottoms out.